src/app/services/tasksService/tasks.service.ts

```diff
@@ -80,12 +89,31 @@ export class TasksService {
       }, 1000);
     });
   }
-
   getAllItems(): Observable<ActionItem[]> {
     return new Observable(observer => {
       setTimeout(() => {
         observer.next(this.dataTable);
       }, 1000);
     });
   }
+  getActionItem(itemId: string): Observable<ActionItem> | undefined {
+    const actionItem = this.dataTable.find(({ id }) => id === itemId);
+    if (actionItem) {
+      return new Observable(observer => {
+        setTimeout(() => {
+          observer.next(actionItem);
+        }, 1000);
+      });
+    } else {
+      return new Observable(observer => {
```

Comment:
**NOTE:** In this case, we should probably return Observable with error (using eg. `return throwError(err);` and do proper error handling.